File: backend/app/usecase/get_job.py
from app.repository.job_repository import JobRepository
from app.repository.cv_repository import CvRepository
from app.domain.errors import NotFoundError


class GetJob:
    def __init__(self, repo: JobRepository, cv_repo: CvRepository):
        self.repo = repo
        self.cv_repo = cv_repo

    async def execute(self, job_id: str) -> dict:
        job = await self.repo.get_by_id(job_id)
        if job is None:
            raise NotFoundError("job_not_found")
        return {"job": job.to_json()}

File: frontend/lib/domain/usecases/get_job.dart
import 'package:injectable/injectable.dart';
import '../models/job.dart';
import '../repositories/job_repository.dart';

@Injectable()
class GetJob {
  const GetJob(this._repository);

  final JobRepository _repository;

  Future<Job> call(String jobId) => _repository.getJob(jobId);
}
